code/modules/mob/living/simple_animal/hostile/terror_spiders/objects.dm

```diff
@@ -0,0 +1,341 @@
+
+// --------------------------------------------------------------------------------
+// ----------------- TERROR SPIDERS: EGGS (USED BY NURSE AND QUEEN TYPES) ---------
+// --------------------------------------------------------------------------------
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/DoLayTerrorEggs(var/lay_type, var/lay_number, var/lay_crawl)
+	stop_automated_movement = 1
+	var/obj/effect/spider/terror_eggcluster/C = new /obj/effect/spider/terror_eggcluster(get_turf(src))
+	C.spiderling_type = lay_type
+	C.spiderling_number = lay_number
+	C.spiderling_ventcrawl = lay_crawl
+	C.faction = faction
+	C.spider_myqueen = spider_myqueen
+	C.master_commander = master_commander
+	C.ai_playercontrol_allowingeneral = ai_playercontrol_allowingeneral
+	C.enemies = enemies
+	//In future, might use something like: C.name = initial(lay_type.egg_name)
+	//However, right now, that doesn't work.
+	if (spider_growinstantly)
+		C.amount_grown = 250
+		C.spider_growinstantly = 1
+	spawn(10)
+		stop_automated_movement = 0
+	if (spider_queen_declared_war)
+		C.spider_queen_declared_war = 1
+
+
+/obj/effect/spider/terror_eggcluster
```

Comment:
this is a lottttt of copy-paste from the regular spider egg cluster; may want to make it a subtype of regular egg clusters and add the new vars where needed.